apis/v1/object_reference_types.go

```diff
@@ -145,3 +145,36 @@ type BackendObjectReference struct {
 	// +optional
 	Port *PortNumber `json:"port,omitempty"`
 }
+
+// ObjectReference identifies an API object including its namespace.
+//
+// The API object must be valid in the cluster; the Group and Kind must
+// be registered in the cluster for this reference to be valid.
+//
+// References to objects with invalid Group and Kind are not valid, and must
+// be rejected by the implementation, with appropriate Conditions set
+// on the containing object.
+type ObjectReference struct {
+	// Group is the group of the referent. For example, "gateway.networking.k8s.io".
+	// When unspecified or empty string, core API group is inferred.
+	Group Group `json:"group"`
+
+	// Kind is kind of the referent. For example "ConfigMap" or "Service".
+	Kind Kind `json:"kind"`
+
+	// Name is the name of the referent.
+	Name ObjectName `json:"name"`
+
+	// Namespace is the namespace of the referenced object. When unspecified, the local
+	// namespace is inferred.
+	//
+	// Note that when a namespace different than the local namespace is specified,
+	// a ReferenceGrant object is required in the referent namespace to allow that
+	// namespace's owner to accept the reference. See the ReferenceGrant
+	// documentation for details.
+	//
+	// Support: Core
+	//
+	// +optional
+	Namespace *Namespace `json:"namespace,omitempty"`
```

Comment:
a namespace with an empty name is an error ""?

Reply:
We use type aliases heavily for common validation, including in this case. So nil is valid (pointer type) but "" is not.

https://github.com/kubernetes-sigs/gateway-api/blob/12d50fe83cc1c8ac050c283f3054d6c513b6d404/apis/v1/shared_types.go#L598-L601